include/umf/memory_pool_ops.h

```diff
@@ -17,37 +17,108 @@
 extern "C" {
 #endif
 
-/// \brief This structure comprises function pointers used by corresponding umfPool*
+///
+/// @brief This structure comprises function pointers used by corresponding umfPool*
 /// calls. Each memory pool implementation should initialize all function
 /// pointers.
+///
 struct umf_memory_pool_ops_t {
     /// Version of the ops structure.
-    /// Should be initialized using UMF_VERSION_CURRENT
+    /// Should be initialized using UMF_VERSION_CURRENT.
     uint32_t version;
 
     ///
-    /// \brief Initializes memory pool.
-    /// \param providers array of memory providers that will be used for coarse-grain allocations.
+    /// @brief Initializes memory pool.
+    /// @param providers array of memory providers that will be used for coarse-grain allocations.
     ///        Should contain at least one memory provider.
-    /// \param numProvider number of elements in the providers array
-    /// \param params pool-specific params
-    /// \param pool [out] returns pointer to the pool
-    /// \return UMF_RESULT_SUCCESS on success or appropriate error code on failure.
+    /// @param numProvider number of elements in the providers array
+    /// @param params pool-specific params
+    /// @param pool [out] returns pointer to the pool
```

Comment:
maybe we need to describe better that returned `pool` pointer (which is not equal to the pool handle) points to the pool-specific private data that is used as an input parameter for subsequent `malloc`, `free`, etc. callbacks of that pool.

@pbalcer, @igchor, @bratpiorka what do you think?

Reply:
I agree, but remember that we have plans to improve all the documentation later anyway - this PR focues on generation. BTW, what I don't like about the oneAPI documentation is that it often focuses mainly on parameters rather than helping the user understand what the function does. So in addition to this kind of API documentation, we should also publish an overview (what we have in our HLD) and a user guide.

Reply:
> we should also publish an overview (what we have in our HLD) and a user guide.

But in a separate PR, right?

Reply:
right